scalafmt-tests/src/test/resources/test/IndentOperator.stat

```diff
@@ -2906,3 +2906,159 @@ object a {
     b
   }
 }
+<<< enclosedAloneOnly, if
+indentOperator.exemptScope = enclosedAlone
+===
+object a {
+  if (
+    a &&
+    b
+  ) {
+    a &&
+    b
+  } else
+    a &&
+    b
+}
+>>>
+object a {
+  if (
+    a &&
+    b
+  ) {
+    a &&
+    b
+  } else
+    a &&
+      b
+}
+<<< enclosedAloneOnly, while
+indentOperator.exemptScope = enclosedAlone
+===
+object a {
+  while (
+    a &&
+    b
+  ) {
+    a &&
+    b
+  }
+  while (
+    a &&
+    b
+  )
+    a &&
+    b
+}
+>>>
+object a {
+  while (
+    a &&
+    b
+  ) {
+    a &&
+    b
+  }
+  while (
+    a &&
+    b
+  )
+    a &&
+      b
+}
+<<< enclosedAloneOnly, apply
+indentOperator.exemptScope = enclosedAlone
+===
+object a {
+  foo(
+    a &&
+    b
+  )
+  foo {
+    a &&
+    b
+  }
+  foo(
+      a &&
+      b,
+      a &&
+      b
+  )
+  foo(
+      a &&
+      b
+  )(
+      a &&
+      b
+  )
+}
+>>>
+object a {
+  foo(
+    a &&
+    b
+  )
+  foo {
+    a &&
+    b
+  }
+  foo(
+    a &&
+      b,
+    a &&
+      b
+  )
```

Comment:
Theses would also be
```suggestion
  foo(
    a &&
    b,
    a &&
    b
  )
```
Even when there are multiple parameters, the binary operator chain is alone *in their actual parameter*.

Reply:
what's an actual parameter?

Reply:
and if you don't need them to be enclosed (which one parameter gives you but multiple do not), why is being alone in the case body (which is what started this conversation) not included?

Reply:
An actual parameter is an argument at call site. As opposed to a formal parameter, which is the parameter at definition site.

> and if you don't need them to be enclosed (which one parameter gives you but multiple do not), why is being alone in the case body (which is what started this conversation) not included?

Hum actually being alone in a `case` body should be included as well, yes.

The "alone" condition is not really about being *enclosed*. It's about avoiding things like
```scala
val foo = {
  doSomethingFirst()
  thenReturnThis ||
  that
}
```
because there it's extremely confusing.

Reply:
ok, now I'm confused. @ekrich, didn't you say that the recent change to infix as entire case body was not welcomed?

@sjrd any other cases? so it's not about having delimiters but being alone? then what about
```
if (...)
  a &&
  b
```
indent or not? it's alone in `then` body. or method body? or yield body?

or
```
// infix retrieves a lambda, and it's alone
(foo get
  bar)(arg)
```

Reply:
> @sjrd any other cases? so it's not about having delimiters but being alone? then what about
> 
> ```
> if (...)
>   a &&
>   b
> ```
> 
> indent or not? it's alone in `then` body. or method body? or yield body?

Our coding style would mandate `{}` around the then branch (because it has more than one line; yes I know scalafmt won't do this for me ;-) ), so this is a case that cannot happen for us. If I go beyond our own style, and we accept this as valid, I think it should be considered alone in the `then` body, yes. I think this is only way that it can decently generalize to braceless syntax anyway.

> 
> or
> 
> ```
> // infix retrieves a lambda, and it's alone
> (foo get
>   bar)(arg)
> ```

Good question. I don't think I would get in that situation, given my coding habits. I don't have a good answer for this one. I would say do whatever is simpler to implement. 🤷‍♂️

Reply:
> Our coding style would mandate `{}` around the then branch (because it has more than one line; yes I know scalafmt won't do this for me ;-) )...

i implemented this rewrite recently: https://scalameta.org/scalafmt/docs/configuration.html#inserting-braces